src/wire/ndiscoption.rs

```diff
@@ -0,0 +1,703 @@
+use core::fmt;
+use byteorder::{NetworkEndian, ByteOrder};
+
+use {Error, Result};
+use time::Duration;
+use wire::{EthernetAddress, Ipv6Address, Ipv6Packet, Ipv6Repr};
+
+enum_with_unknown! {
+    /// NDISC Option Type
+    pub doc enum Type(u8) {
+        /// Source Link-layer Address
+        SourceLinkLayerAddr = 0x1,
+        /// Target Link-layer Address
+        TargetLinkLayerAddr = 0x2,
+        /// Prefix Information
+        PrefixInformation   = 0x3,
+        /// Redirected Header
+        RedirectedHeader    = 0x4,
+        /// MTU
+        Mtu                 = 0x5
+    }
+}
+
+impl fmt::Display for Type {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match self {
+            &Type::SourceLinkLayerAddr => write!(f, "source link-layer address"),
+            &Type::TargetLinkLayerAddr => write!(f, "target link-layer address"),
+            &Type::PrefixInformation   => write!(f, "prefix information"),
+            &Type::RedirectedHeader    => write!(f, "redirected header"),
+            &Type::Mtu                 => write!(f, "mtu"),
+            &Type::Unknown(id) => write!(f, "{}", id)
+        }
+    }
+}
+
+bitflags! {
+    pub struct PrefixInfoFlags: u8 {
+        const ON_LINK  = 0b10000000;
+        const ADDRCONF = 0b01000000;
+    }
+}
+
+/// A read/write wrapper around an [NDISC Option].
+///
+/// [NDISC Option]: https://tools.ietf.org/html/rfc4861#section-4.6
+#[derive(Debug, PartialEq)]
+pub struct NdiscOption<T: AsRef<[u8]>> {
+    buffer: T
+}
+
+// Format of an NDISC Option
+//
+// +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+// |     Type      |    Length     |              ...              |
+// +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+// ~                              ...                              ~
+// +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+//
+// See https://tools.ietf.org/html/rfc4861#section-4.6 for details.
+mod field {
```

Comment:
Wasn't sure how to break up the groupings of constants. If this is not readable, I'm more than happy to iterate on this.